### ISSUE TITLE
Fetch emoji code/codepoints from server and store under realm and use it instead of using static codes.

### ISSUE THREAD

**Issue by jainkuniya** [zulip/zulip-mobile#2956]
Related discussion: https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/Emoji.20sync

@zulipbot claim

**Comment by gnprice** [comment]
One example mismatch I noticed while fixing #3129 (described in a9f2666d6):

```
There's still a bug that remains, which is that we think `:0:`
etc. are names for the 10 single-digit emoji (in addition to the
spelled-out `￼:zero:`), but the server seems to disagree; as a result
autocomplete and adding a reaction don't work.
```

**Comment by gnprice** [comment]
This is a followup to #2868.

Doing this is expected to resolve #3128.

**Comment by gnprice** [comment]
Marking this as a priority. See https://github.com/zulip/zulip-mobile/issues/3128#issuecomment-802289882 , and the linked [chat discussion](https://chat.zulip.org/#narrow/stream/243-mobile-team/topic/emoji.20list/near/1135704).

**Comment by chrisbobbe** [comment]
We should do another ad-hoc fix for the `:slight_smile:` to `:smile` change, then remove the P1 label.

**Comment by WesleyAC** [comment]
Removing P1, since I filed #4640. It's not merged yet, but I'd rather not forget to remove the label, so just doing it now.

**Comment by gnprice** [comment]
> Removing P1, since I filed #4640. It's not merged yet, but I'd rather not forget to remove the label, so just doing it now.

It's all come out the same because #4640 is merged now, but for similar things in the future: if something's P1 to do, I'd like to make sure there's something open for it that has a P1 label until it's actually done. Forgetting to remove a P1 label is a much lower risk than forgetting to do the thing, because the P1 label itself keeps it visible as a thing we should take care of.

In this instance, one good option would have been to add a P1 label on #4640, and then remove it on this issue.

**Comment by WesleyAC** [comment]
Bumping the priority on this again, since we expect to do a big emoji update on the server side soon (:crossed_fingers:), and it'll be very awkward to not have this then.

**Comment by chrisbobbe** [comment]
Wordle is broken on mobile on CZO 😛; see https://chat.zulip.org/#narrow/stream/97-off-topic/topic/Wordle/near/1322606

**Comment by chrisbobbe** [comment]
Just filed https://github.com/zulip/zulip/issues/21037; marking as blocked on that.

**Comment by gnprice** [comment]
There's now a draft server API for this, implemented in the open PR zulip/zulip#22067.

So this is ripe for us to implement. The implementation will help provide feedback on the API, before merging the server PR. Then after the server change is merged, we can merge our client implementation.

**Comment by gnprice** [comment]
@chrisbobbe This was fixed by #5464, right? Or is there a portion of this issue that remains open?

If there's still something to track here, let's open new issues for that.

**Comment by chrisbobbe** [comment]
I believe this is all taken care of; I can't think of anything else we need for this, and I don't see anything that looks unfinished in the discussion above (using GitHub for iOS).
